### PR TITLE
Esp32 s3 support

### PR DESCRIPTION
**Pull request by me-no-dev**

### CODE REVIEW

**Comment by VojtechBartoska** [comment]
Closes https://github.com/espressif/arduino-esp32/issues/6030 (Rainmaker support for ESP32-S2/C3 and S3)

File: configs/defconfig.esp32 [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+CONFIG_LWIP_DHCP_OPTIONS_LEN=128
+CONFIG_LWIP_TCP_SYNMAXRTX=6
+CONFIG_LWIP_TCP_MSS=1436
+CONFIG_LWIP_TCP_RTO_TIME=3000
+CONFIG_LWIP_TCPIP_TASK_STACK_SIZE=2560
+CONFIG_LWIP_TCPIP_TASK_AFFINITY_CPU0=y
+CONFIG_LWIP_SNTP_MAX_SERVERS=3
+CONFIG_LWIP_DHCP_GET_NTP_SRV=y
+# CONFIG_MBEDTLS_ASYMMETRIC_CONTENT_LEN is not set
+# CONFIG_MBEDTLS_HARDWARE_SHA is not set
```
Comment (igrr) [inline configs/defconfig.esp32:65]: CONFIG_MBEDTLS_HARDWARE_SHA should be enabled (https://github.com/espressif/arduino-esp32/issues/976)

File: configs/defconfig.esp32s2 [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+CONFIG_LWIP_TCP_SYNMAXRTX=6
+CONFIG_LWIP_TCP_MSS=1436
+CONFIG_LWIP_TCP_RTO_TIME=3000
+CONFIG_LWIP_TCPIP_TASK_STACK_SIZE=2560
+CONFIG_LWIP_TCPIP_TASK_AFFINITY_CPU0=y
+CONFIG_LWIP_SNTP_MAX_SERVERS=3
+CONFIG_LWIP_DHCP_GET_NTP_SRV=y
+# CONFIG_MBEDTLS_ASYMMETRIC_CONTENT_LEN is not set
+# CONFIG_MBEDTLS_HARDWARE_MPI is not set
+# CONFIG_MBEDTLS_HARDWARE_SHA is not set
```
Comment (igrr) [inline configs/defconfig.esp32s2:67]: These 2 should be enabled, i think: CONFIG_MBEDTLS_HARDWARE_MPI & CONFIG_MBEDTLS_HARDWARE_SHA

File: configs/defconfig.esp32s3 [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+CONFIG_LWIP_TCP_SYNMAXRTX=6
+CONFIG_LWIP_TCP_MSS=1436
+CONFIG_LWIP_TCP_RTO_TIME=3000
+CONFIG_LWIP_TCPIP_TASK_STACK_SIZE=2560
+CONFIG_LWIP_TCPIP_TASK_AFFINITY_CPU0=y
+CONFIG_LWIP_SNTP_MAX_SERVERS=3
+CONFIG_LWIP_DHCP_GET_NTP_SRV=y
+# CONFIG_MBEDTLS_ASYMMETRIC_CONTENT_LEN is not set
+# CONFIG_MBEDTLS_HARDWARE_MPI is not set
+# CONFIG_MBEDTLS_HARDWARE_SHA is not set
```
Comment (igrr) [inline configs/defconfig.esp32s3:69]: likewise

**Pull request merged**